extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -119,23 +122,77 @@ export class ProjectsController {
 		this.refreshProjectsTree();
 	}
 
-	public async buildProject(treeNode: BaseProjectTreeItem): Promise<void> {
+	/**
+	 * Builds a project, producing a dacpac
+	 * @param treeNode a treeItem in a project's hierarchy, to be used to obtain a Project
+	 * @returns path of the built dacpac
+	 */
+	public async buildProject(treeNode: BaseProjectTreeItem): Promise<string>;
+	/**
+	 * Builds a project, producing a dacpac
+	 * @param project Project to be built
+	 * @returns path of the built dacpac
+	 */
+	public async buildProject(project: Project): Promise<string>;
+	public async buildProject(context: Project | BaseProjectTreeItem): Promise<string | undefined> {
+		const project: Project = context instanceof Project ? context : this.getProjectContextFromTreeNode(context);
+
 		// Check mssql extension for project dlls (tracking issue #10273)
 		await this.buildHelper.createBuildDirFolder();
 
-		const project = this.getProjectContextFromTreeNode(treeNode);
 		const options: DotNetCommandOptions = {
 			commandTitle: 'Build',
 			workingDirectory: project.projectFolderPath,
 			argument: this.buildHelper.constructBuildArguments(project.projectFilePath, this.buildHelper.extensionBuildDirPath)
 		};
-		await this.netCoreTool.runDotnetCommand(options);
+		try {
+			await this.netCoreTool.runDotnetCommand(options);
+
+			return path.join(project.projectFolderPath, 'bin', 'Debug', `${project.projectFileName}.dacpac`);
+		}
+		catch (err) {
+			this.apiWrapper.showErrorMessage(constants.projBuildFailed(utils.getErrorMessage(err)));
+			return;
+		}
 	}
 
-	public deploy(treeNode: BaseProjectTreeItem): void {
-		const project = this.getProjectContextFromTreeNode(treeNode);
-		const deployDatabaseDialog = new DeployDatabaseDialog(this.apiWrapper, project);
+	/**
+	 * Builds and deploys a project
+	 * @param treeNode a treeItem in a project's hierarchy, to be used to obtain a Project
+	 */
+	public async deployProject(treeNode: BaseProjectTreeItem): Promise<DeployDatabaseDialog>;
+	/**
+	 * Builds and deploys a project
+	 * @param project Project to be built and deployed
+	 */
+	public async deployProject(project: Project): Promise<DeployDatabaseDialog>;
```

Comment:
Why do you have these two other signatures? Just the base one is enough with the OR'd type